File: Medium/973-k-closest-points-to-origin/try-5.py
/*
 * Problem #973: K Closest Points to Origin
 * Difficulty: Medium
 * Submission: Try 5
 * status: Accepted
 * Language: python3
 * Date: 17/12/2024, 11:47:46
 * Link: https://leetcode.com/problems/k-closest-points-to-origin/
 */

class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        distance = [ [p[0]**2 + p[1]**2 , ind ] for ind , p in enumerate(points)]
        heapify(distance)
        return [points[heappop(distance)[-1]] for _ in range(k)]
        

File: Easy/27-remove-element/try-5.cpp
/*
 * Problem #27: Remove Element
 * Difficulty: Easy
 * Submission: Try 5
 * status: Accepted
 * Language: cpp
 * Date: 21/11/2024, 22:08:21
 * Link: https://leetcode.com/problems/remove-element/
 */

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int ip=0,pt=nums.size()-1;
    
        while(ip<=pt){
            if(nums[ip]==val){
                swap(nums[ip],nums[pt--]);
                continue;
            }
            ip++;
        }
        return ip++;
    }

};
